src/Handler/DatabaseHandler.php

```diff
@@ -0,0 +1,304 @@
+<?php
+/**
+ * Part of the Joomla Framework Session Package
+ *
+ * @copyright  Copyright (C) 2005 - 2015 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE
+ */
+
+namespace Joomla\Session\Handler;
+
+use Joomla\Database\DatabaseDriver;
+use Joomla\Session\HandlerInterface;
+
+/**
+ * Database session storage handler
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+class DatabaseHandler implements HandlerInterface
+{
+	/**
+	 * Database connector
+	 *
+	 * @var    \Joomla\Database\DatabaseDriver
+	 * @since  __DEPLOY_VERSION__
+	 */
+	private $db;
+
+	/**
+	 * Flag whether gc() has been called
+	 *
+	 * @var    boolean
+	 * @since  __DEPLOY_VERSION__
+	 */
+	private $gcCalled = false;
+
+	/**
+	 * Lifetime for garbage collection
+	 *
+	 * @var    integer
+	 * @since  __DEPLOY_VERSION__
+	 */
+	private $gcLifetime;
+
+	/**
+	 * Constructor
+	 *
+	 * @param   DatabaseDriver  $db  Database connector
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function __construct(DatabaseDriver $db)
+	{
+		$this->db = $db;
+	}
+
+	/**
+	 * Close the session
+	 *
+	 * @return  boolean  True on success, false otherwise
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function close()
+	{
+		if ($this->gcCalled)
+		{
+			$this->gcCalled   = false;
+			$this->gcLifetime = null;
+
+			$query = $this->db->getQuery(true)
+				->delete($this->db->quoteName('#__session'))
+				->where($this->db->quoteName('time') . ' < ' . $this->db->quote((int) $this->gcLifetime));
+
+			// Remove expired sessions from the database.
+			$this->db->setQuery($query)->execute();
+		}
+
+		$this->db->disconnect();
```

Comment:
At this point the session is effectively closed out so a connection isn't needed any longer.  If it gets reconnected, `DatabaseDriver` should handle reconnecting already.